nomad/leader.go

```diff
@@ -625,6 +627,29 @@ func (s *Server) publishJobSummaryMetrics(stopCh chan struct{}) {
 								Value: name,
 							},
 						}
+
+						if strings.Contains(summary.JobID, "/dispatch-") {
+							jobInfo := strings.Split(summary.JobID, "/dispatch-")
+							labels = append(labels, metrics.Label{
+								Name:  "parent_id",
+								Value: jobInfo[0],
+							}, metrics.Label{
+								Name:  "dispatch_id",
+								Value: jobInfo[1],
+							})
+						}
+
+						if strings.Contains(summary.JobID, "/periodic-") {
+							jobInfo := strings.Split(summary.JobID, "/periodic-")
+							labels = append(labels, metrics.Label{
+								Name:  "parent_id",
+								Value: jobInfo[0],
+							}, metrics.Label{
+								Name:  "periodic_id",
+								Value: jobInfo[1],
+							})
```

Comment:
Two points overall: 
- Consider adding a helper function where there is code duplication
- Don't assume the length of the array is what is as expected, first check before indexing.